[v2.2.1] Auth0Plugin is incompatible with Auth0VueClient (exactOptionalPropertyTypes: true)

### Checklist

- [X] The issue can be reproduced in the [auth0-vue sample app](https://github.com/auth0-samples/auth0-vue-samples/tree/master/01-Login) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/auth0-vue#readme), [Examples](https://github.com/auth0/auth0-vue/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/auth0-vue/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/auth0-vue/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-vue/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

After updating to v2.2.1, my typscript compiler complains when assigning a value of type `Auth0Plugin` to a variable defined as type `Auth0VueClient`.

Error:
```ts
plugins/1.auth0.server.ts:10:5 - error TS2322: Type 'Auth0Plugin' is not assignable to type 'Auth0VueClient'.
  The types of 'user.value' are incompatible between these types.
    Type '{ readonly [x: string]: any; readonly name?: string | undefined; readonly given_name?: string | undefined; readonly family_name?: string | undefined; readonly middle_name?: string | undefined; ... 15 more ...; readonly sub?: string | undefined; } | undefined' is not assignable to type 'User | undefined'.
      Type '{ readonly [x: string]: any; readonly name?: string | undefined; readonly given_name?: string | undefined; readonly family_name?: string | undefined; readonly middle_name?: string | undefined; ... 15 more ...; readonly sub?: string | undefined; }' is not assignable to type 'User' with 'exactOptionalPropertyTypes: true'. Consider adding 'undefined' to the types of the target's properties.
        Types of property 'name' are incompatible.
          Type 'string | undefined' is not assignable to type 'string'.
            Type 'undefined' is not assignable to type 'string'.
```

The types of the values `user` of the interface `Auth0VueClient` and the class `Auth0Plugin` are incompatible in that the class defines it as:
```ts
export declare class Auth0Plugin implements Auth0VueClient {
    private clientOptions;
    private pluginOptions?;
    private _client;
    private _isLoading;
    private _isAuthenticated;
    private _user;
    private _idTokenClaims;
    private _error;
    isLoading: Readonly<Ref<boolean>>;
    isAuthenticated: Readonly<Ref<boolean>>;
    user: Readonly<Ref<{
        readonly [x: string]: any;
        readonly name?: string | undefined;
        readonly given_name?: string | undefined;
        readonly family_name?: string | undefined;
        readonly middle_name?: string | undefined;
        readonly nickname?: string | undefined;
        readonly preferred_username?: string | undefined;
        readonly profile?: string | undefined;
        readonly picture?: string | undefined;
        readonly website?: string | undefined;
        readonly email?: string | undefined;
        readonly email_verified?: boolean | undefined;
        readonly gender?: string | undefined;
        readonly birthdate?: string | undefined;
        readonly zoneinfo?: string | undefined;
        readonly locale?: string | undefined;
        readonly phone_number?: string | undefined;
        readonly phone_number_verified?: boolean | undefined;
        readonly address?: string | undefined;
        readonly updated_at?: string | undefined;
        readonly sub?: string | undefined;
    } | undefined>>;
```
... but the interface refers to the class `User` which is defined as:
```ts
export declare class User {
    name?: string;
    given_name?: string;
    family_name?: string;
    middle_name?: string;
    nickname?: string;
    preferred_username?: string;
    profile?: string;
    picture?: string;
    website?: string;
    email?: string;
    email_verified?: boolean;
    gender?: string;
    birthdate?: string;
    zoneinfo?: string;
    locale?: string;
    phone_number?: string;
    phone_number_verified?: boolean;
    address?: string;
    updated_at?: string;
    sub?: string;
    [key: string]: any;
}
```

When having the option `exactOptionalPropertyTypes` set to `true`, an optional property is different to an optional property which can also be undefined.

### Reproduction

1. Setting `exactOptionalPropertyTypes` to `true` in your Typescript configuration
2. Try to let this statement compile:
```ts
const foo = (plugin: Auth0Plugin) : Auth0VueClient => plugin
```

### Additional context

_No response_

### auth0-vue version

2.2.1

### Vue version

3.3.4

### Which browsers have you tested in?

Other

--- a/src/plugin.ts
+++ b/src/plugin.ts
@@ -1,5 +1,5 @@
 import type { App, Ref } from 'vue';
-import { reactive, readonly, ref } from 'vue';
+import { ref } from 'vue';
 import type { Router } from 'vue-router';
 import type {
   AppState,

--- a/src/plugin.ts
+++ b/src/plugin.ts
@@ -1,5 +1,5 @@
 import type { App, Ref } from 'vue';
-import { readonly, ref } from 'vue';
+import { reactive, readonly, ref } from 'vue';
 import type { Router } from 'vue-router';
 import type {
   AppState,
@@ -62,17 +62,11 @@ export const client: Ref<Auth0VueClient> = ref(
  */
 export class Auth0Plugin implements Auth0VueClient {
   private _client!: Auth0Client;
-  private _isLoading: Ref<boolean> = ref(true);
-  private _isAuthenticated: Ref<boolean> = ref(false);
-  private _user: Ref<User | undefined> = ref({});
-  private _idTokenClaims = ref<IdToken>();
-  private _error = ref<Error | null>(null);
-
-  isLoading = readonly(this._isLoading);
-  isAuthenticated = readonly(this._isAuthenticated);
-  user = readonly(this._user);
-  idTokenClaims = readonly(this._idTokenClaims);
-  error = readonly(this._error);
+  public isLoading: Ref<boolean> = ref(true);
+  public isAuthenticated: Ref<boolean> = ref(false);
+  public user: Ref<User | undefined> = ref({});
+  public idTokenClaims = ref<IdToken | undefined>();
+  public error = ref<Error | null>(null);
 
   constructor(
     private clientOptions: Auth0VueClientOptions,
@@ -194,19 +188,19 @@ export class Auth0Plugin implements Auth0VueClient {
   }
 
   private async __refreshState() {
-    this._isAuthenticated.value = await this._client.isAuthenticated();
-    this._user.value = await this._client.getUser();
-    this._idTokenClaims.value = await this._client.getIdTokenClaims();
-    this._isLoading.value = false;
+    this.isAuthenticated.value = await this._client.isAuthenticated();
+    this.user.value = await this._client.getUser();
+    this.idTokenClaims.value = await this._client.getIdTokenClaims();
+    this.isLoading.value = false;
   }
 
   private async __proxy<T>(cb: () => T, refreshState = true) {
     let result;
     try {
       result = await cb();
-      this._error.value = null;
+      this.error.value = null;
     } catch (e) {
-      this._error.value = e as Error;
+      this.error.value = e as Error;
       throw e;
     } finally {
       if (refreshState) {

